Mathlib/Combinatorics/SimpleGraph/Finite.lean

```diff
@@ -435,4 +435,79 @@ theorem card_commonNeighbors_top [DecidableEq V] {v w : V} (h : v ≠ w) :
 
 end Finite
 
+section Support
+
+variable [DecidableEq V] {s : Set V} [DecidablePred (· ∈ s)]
+  [Fintype V] {G : SimpleGraph V} [DecidableRel G.Adj]
+
+instance : DecidablePred (· ∈ G.support) :=
+  inferInstanceAs <| DecidablePred (· ∈ { v | ∃ w, G.Adj v w })
+
+theorem map_edgeFinset_induce :
+    (G.induce s).edgeFinset.map (Embedding.subtype s).sym2Map
+      = G.edgeFinset ∩ s.toFinset.sym2 := by
+  simp_rw [Finset.ext_iff, Sym2.forall, mem_inter, mk_mem_sym2_iff, mem_map, Sym2.exists,
+    Set.mem_toFinset, mem_edgeSet, comap_adj, Embedding.sym2Map_apply, Embedding.coe_subtype,
+    Sym2.map_pair_eq, Sym2.eq_iff]
+  intro v w
+  constructor
+  · rintro ⟨x, y, hadj, ⟨hv, hw⟩ | ⟨hw, hv⟩⟩
+    all_goals rw [← hv, ← hw]
+    · exact ⟨hadj, x.prop, y.prop⟩
+    · exact ⟨hadj.symm, y.prop, x.prop⟩
+  · intro ⟨hadj, hv, hw⟩
+    use ⟨v, hv⟩, ⟨w, hw⟩, hadj
+    tauto
+
+omit [DecidableEq V] in
+lemma edgeFinset_subset_sym2_of_support_subset (h : G.support ⊆ s) :
```

Comment:
It looks sensible to me to move this lemma up by two, and put the `[DecidableEq V]` variable declaration after it.